packages/api/src/contracts/survey.repository.ts

```diff
@@ -0,0 +1,9 @@
+import type { Survey, UserSurvey } from '@coderscamp/shared/models';
+
+export abstract class SurveyRepositoryPort {
```

Comment:
Very good! But i think saveAndUpdateRelatedUser is not necessary. What is a use case for this method?

Reply:
The idea is just save the initial form in one go. There is no special reason for doing so this way as from what I read Prisma makes 2 separate queries anyway.

Reply:
So if 2 separate queries - no optimization here, it's better to do not have coupling between Survey and User just for... what :) ? We don't have any advantages, but we have blurred survey module with user concept.